core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -518,6 +518,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _taskScheduler.postStartHook()
     _env.metricsSystem.registerSource(new DAGSchedulerSource(dagScheduler))
     _env.metricsSystem.registerSource(new BlockManagerSource(_env.blockManager))
+    _executorAllocationManager.foreach { e =>
```

Comment:
The model for the other metric sources seems to be that the source wraps the object.  What's the reason for having the source inside of the ExecutorAllocationManager in this case?